lms/djangoapps/lti_provider/tasks.py

```diff
@@ -0,0 +1,79 @@
+"""
+Asynchronous tasks for the LTI provider app.
+"""
+
+from django.core.exceptions import ObjectDoesNotExist
+import logging
+
+from lms import CELERY_APP
+from lti_provider.models import GradedAssignment
+import lti_provider.outcomes
+from lti_provider.views import parse_course_and_usage_keys
+
+log = logging.getLogger("edx.lti_provider")
+
+
+@CELERY_APP.task
+def send_outcome(points_possible, points_earned, user_id, course_id, usage_id):
+    """
+    Calculate the score for a given user in a problem and send it to the
+    appropriate LTI consumer's outcome service.
+    """
+    if points_earned is None or not points_possible:
+        log.error(
+            "Outcome Service: Invalid Points earned (%s) or points possible (%s)",
+            points_earned, points_possible
+        )
+        return
```

Comment:
Perhaps it makes sense to do this kind of checking at the point where the signal is created as opposed to on the receiving end? It seems like every client would want to ignore score changed events with invalid values for earned, possible, course, or usage.